core/src/kura.rs

```diff
@@ -168,6 +171,12 @@ impl Kura {
                 }
             };
 
+            if failed_to_store_block.is_some() {
+                let error = failed_to_store_block_reason.unwrap();
+                error!(%error, "Failed to write block. Kura will now init again.");
```

Comment:
OK, if we really are going to retry the `init` step, then you can end up in a boot loop, which is terrible. Rather I'd like to have a constant that sets the retry count, tries 1-2 times and panics with an error message.